### ISSUE TITLE
Result pic misbehavior

### ISSUE THREAD

**Issue by maryfries** [jmoenig/Snap#3257]
When I do a script pic, I get this
![Untitled script pic (17)](https://github.com/jmoenig/Snap/assets/6549787/29eefd06-d86b-4321-bade-63a5298c29f4)
but when I do a result pic (clicking the operator), I get this
![Untitled script pic (18)](https://github.com/jmoenig/Snap/assets/6549787/4ba02f7e-4e34-4581-869b-4ef636fe6353)
Visible stepping is _not_ on.
I want a result pic for the whole expression (don't mind the error; I was trying to make an error pic).
Interestingly, it works as expected for other error types:
![Untitled script pic (19)](https://github.com/jmoenig/Snap/assets/6549787/ccffcb50-d239-47ad-b70e-80adc8d33057)

**Comment by maryfries** [comment]
FWIW, `pets` was _not_ defined.

**Comment by brianharvey** [comment]
I agree with you wanting a picture of the entire expression you clicked on.  But the "interestingly" part at the bottom isn't surprising, for two reasons: (1) "NaN" isn't an error; it doesn't have that "Hmm..." in front of it.  It's considered a perfectly good numeric-type value.  So the two cases aren't comparable.  (2) In any case, the ITEM part of that expression worked fine, and gave "apple" as its result; it's the + that generated the error, in response to 3+apple.

About your second message, Snap<i>!</i> variables are initialized to zero.
